tests/components/wireguard/test_sensor.py

```diff
@@ -0,0 +1,64 @@
+"""Test the WireGuard sensor platform."""
+from unittest.mock import MagicMock
+
+from homeassistant.components.sensor import SensorDeviceClass
+from homeassistant.components.wireguard.const import DOMAIN
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_UNIT_OF_MEASUREMENT,
+    UnitOfInformation,
+)
+from homeassistant.core import HomeAssistant, State
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import MockConfigEntry
+
+
+async def test_entity_registry(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    config_entry: MockConfigEntry,
+    coordinator_client: MagicMock,
+) -> None:
+    """Tests that the devices are registered in the entity registry."""
+    config_entry.add_to_hass(hass)
+    coordinator_client.side_effect = None
+
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    await coordinator.async_refresh()
+    await hass.async_block_till_done()
```

Comment:
Parts of this which are reused in every test can be outsourced in a function as well. (Most integrations call it `init_integration` or `async_init_integration`)